### PR TITLE
Tutorial

### PR DESCRIPTION
**Pull request by Paul-Ferrell**

### CODE REVIEW

File: docs/install.rst [hunk 1]
```diff
@@ -153,3 +171,22 @@ It comes with a couple of caveats:
...
+===================
+
+Pavilion uses the ``module`` command to load modules for tests. It will work
+with either lmod or the tcl based 'environment modules' systems. This is
+generally only needed if your cluster/s have a complex software environment
+that supports multiple compilers and conflicting builds of libraries.
+
+It is assumed that the module environment is set up before you run Pavilion. If
+you need to set up this environment separately on allocations, use the
+'env_setup' option in the general Pavilion configuration to add the commands
```
Comment (nicholas-sly) [inline docs/install.rst]: Link to the appropriate place in the documentation for this.

File: docs/tutorials/writing_a_test.rst [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+
+Any tests found in your ``~/.pavilion`` directory will take precedence over
+those found in the general Pavilion config directory.
+
+Writing a Test
+--------------
+
+.. _supermagic: https://github.com/hpc/supermagic
+
+We're going to use the `supermagic`_ as our example.
```
Comment (nicholas-sly) [inline docs/tutorials/writing_a_test.rst]: Either remove the word 'the' or add the word 'test' after supermagic.

File: docs/tutorials/writing_a_test.rst [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+.. code-block:: shell
+
+    $ pav show tests
+
+    -----------------+------------------------
+     Name            | Summary
+    -----------------+------------------------
+    supermagic.basic | A basic supermagic run.
+
+If your suite or test is highlighted in red, there was an error in your
```
Comment (nicholas-sly) [inline docs/tutorials/writing_a_test.rst]: '... highlighted in red and/or followed by an asterisk...'

File: docs/tutorials/writing_a_test.rst [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+When building the test Pavilion will run that script in the extracted build
+directory.
+
+Let's try it:
+
+.. code-block:: shell
+
+    $ pav run supermagic.basic
+
+    Test tutorial.basic run 72 building 787aceaa19ac9a21
```
Comment (nicholas-sly) [inline docs/tutorials/writing_a_test.rst]: tutorial.basic or supermagic.basic?

File: docs/tutorials/writing_a_test.rst [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+    Error building test:
+    status BUILD_FAILED - Build returned a non-zero result.
+    For more information, run 'pav log build 72'
+
+Oh no! Our build failed. Let's follow the suggestion, and look at the build
+log for our test. We can also use '``pav cat 72 build.sh``' to output the build
```
Comment (nicholas-sly) [inline docs/tutorials/writing_a_test.rst]: Still want a spack-like `pav cd 72` to take you to the test_run directory of that test.

File: docs/tutorials/writing_a_test.rst [hunk 6]
```diff
@@ -0,0 +1,414 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We tried to build with gcc, but supermagic requires an mpi compiler wrapper.
+We'll have to provide that somehow. Typically that's done with module files.
+So let's modify the build section of our test config to load those modules.
+
+Note:
+  Module loading works with lmod and environment modules (tmod), and
+  assumes the module environment is set up automatically on login. This is
+  covered in more details in the install instructions.
```
Comment (nicholas-sly) [inline docs/tutorials/writing_a_test.rst]: Link to install instructions.

File: docs/tutorials/writing_a_test.rst [hunk 7]
```diff
@@ -0,0 +1,414 @@
...
+
+.. code-block:: yaml
+
+    basic:
+
+        # We'll just configure slurm to use two nodes, and two processes each.
+        # We could also put in a range, or even 'all'.
+        slurm:
+            num_nodes: 2
+            procs_per_node: 2
```
Comment (nicholas-sly) [inline docs/tutorials/writing_a_test.rst]: Should be 'tasks_per_node', unless you've changed that in this pull request.

File: docs/tutorials/writing_a_test.rst [hunk 8]
```diff
@@ -0,0 +1,414 @@
...
+    #SBATCH --tasks-per-node=2
+
+    # Redirect all output to kickoff.log
+    exec >/users/pflarr/.pavilion/working_dir/test_runs/0000020/kickoff.log 2>&1
+
+    # Set the path so we can find the pavilion command that started this test.
+    export PATH=/yellow/usr/projects/hpctools/pflarr/repos/pavilion/bin:${PATH}
+
+    # Point pavilion to the config file that configured it.
+    export PAV_CONFIG_FILE=None
```
Comment (nicholas-sly) [inline docs/tutorials/writing_a_test.rst]: Can pavilion run with no config file?

File: docs/tutorials/writing_a_test.rst [hunk 9]
```diff
@@ -0,0 +1,414 @@
...
+
+    # Perform the sequence of test commands.
+    srun -N 2 -n 4 ./supermagic
+
+There are few things to point out.
+
+1.  The result of a test defaults to the whether run script returns zero. This
+    usually just ends up being the return value of the last of your test
+    commands.
+    If there are critical commands before that make sure to add an
```
Comment (nicholas-sly) [inline docs/tutorials/writing_a_test.rst]: '...before that, make sure...'

File: docs/tutorials/writing_a_test.rst [hunk 10]
```diff
@@ -0,0 +1,414 @@
...
+    ``|| exit 1`` to them. (This isn't needed here).
+2.  Our test script cmd was '``{{sched.test_cmd}} ./supermagic``. The part in
+    double curly braces is a Pavilion variable reference, which our scheduler
+    replaces with an srun command based on our scheduler settings.
+3.  It's important to use '``{{sched.test_cmd}}``'  rather than srun directly.
+    Pavilion tests may run in larger allocations than you request, and this
+    makes sure each test only runs under what it requested.
+4.  The run script is executed from inside our build directory. It's free to
+    create or overwrite any files contained within, as
```
Comment (nicholas-sly) [inline docs/tutorials/writing_a_test.rst]: You seem to have trailed off...

File: lib/pavilion/scriptcomposer.py [hunk 11]
```diff
@@ -175,11 +175,13 @@ def env_change(self, env_dict):
                 new_dict[key] = env_dict[key]
 
         for key, value in env_dict.items():
+
             if value is not None:
                 self._script_lines.append('export {}={}'.format(key, value))
             else:
                 self._script_lines.append('unset {}'.format(key))
 
+
```
Comment (nicholas-sly) [inline lib/pavilion/scriptcomposer.py]: Single line spacing inside of a class.

File: docs/tutorials/writing_a_test.rst (hunk 8 above)
Comment (Paul-Ferrell) [reply docs/tutorials/writing_a_test.rst]: Yes.

File: docs/tutorials/writing_a_test.rst (hunk 5 above)
Comment (Paul-Ferrell) [reply docs/tutorials/writing_a_test.rst]: Are you aware that it isn't possible?

**Pull request merged**